src/anonymous-credentials/presentation-builder.ts

```diff
@@ -492,6 +560,96 @@ export class PresentationBuilder extends Versioned {
       flattenedSchemas.push(flattenedSchema);
     }
 
+    // Create statements and witnesses for each boundedPseudonyms
+    const presentedBoundedPseudonyms: Map<string, IPresentedBoundedPseudonym> = new Map();
+    for (let i = 0; i < this.boundedPseudonyms.length; i++) {
+      // the second parameter is the corresponding attribute index
+      // to do witnessEq.addWitnessRef(sIdx, index);
+      // later when sIdx will be known
+      const witnessEqs: [WitnessEqualityMetaStatement, number][] = [];
+      const attributeBoundPseudonym = this.boundedPseudonyms[i];
+      const attributes: Uint8Array[] = [];
+      for (const [credIdx, attributeNames] of attributeBoundPseudonym.attributeNames.entries()) {
+        // lots of the following is extracted from getRevealedAndUnrevealed and should probably be packaged in a function
+        const cred = this.credentials[credIdx][0];
+        const schema = cred.schema as CredentialSchema;
+        const messages = cred.serializeForSigning();
+        const encoder = schema.encoder;
+        const [names, encodedValues] = encoder.encodeMessageObject(messages);
```

Comment:
You don't need to encode again as its already been done. Cache the encoded values like [here](https://github.com/docknetwork/crypto-wasm-ts/blob/master/src/anonymous-credentials/presentation-builder.ts#L392)

Reply:
I removed it entirely, like you explained here: https://github.com/docknetwork/crypto-wasm-ts/pull/19#discussion_r1206692244